indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -825,7 +824,7 @@ public void fetchAndProcessIfNeeded()
             processQueryGranularity(index);
             processDimensionsSpec(index);
             processMetricsSpec(index);
-            processMultiValuedColumns(index);
+            processMultiValuedDimensions(index);
```

Comment:
```suggestion
            this.multiValuedDimensions = determineMultiValuedDimensions(index);
```

Reply:
Didn't follow this. The `multiValuedDimensions` set is built incrementally by looping over the segments and calling `processMultiValuedDimensions` for `index` on each of them.

Reply:
Ah, you are right. Please leave it as is then.